### PR TITLE
feat(cronjob): Add more properties

### PR DESCRIPTION
**Pull request by AntoineFr**
This PR adds support for the following properties in CronJob:
- `.spec.jobTemplate.spec.activeDeadlineSeconds`
- `.spec.jobTemplate.spec.backoffLimit`
- `.spec.jobTemplate.spec.template.metadata.annotations`
- `.spec.jobTemplate.spec.template.metadata.labels`
- `.spec.jobTemplate.spec.template.spec.topologySpreadConstraints`
- `.spec.jobTemplate.spec.template.spec.securityContext`

### CODE REVIEW

File: application/templates/cronjob.yaml [hunk 1]
```diff
@@ -31,10 +31,17 @@ spec:
...
+{{- if $job.activeDeadlineSeconds }}
+      activeDeadlineSeconds: {{ $job.activeDeadlineSeconds }}
+{{ end }}
       template:
         metadata:
           labels:
           {{- include "application.labels" $ | nindent 12 }}
+          {{- with $job.additionalPodAnnotations }}
+          annotations:
+{{ toYaml . | indent 12 }}
```
Comment (aslafy-z) [inline application/templates/cronjob.yaml:43]:
```suggestion
          annotations: {{ toYaml . | nindent 12 }}
```

File: application/templates/cronjob.yaml [hunk 2]
```diff
@@ -31,10 +31,17 @@ spec:
 {{ end }}
   jobTemplate:
     spec:
+{{- if $job.activeDeadlineSeconds }}
+      activeDeadlineSeconds: {{ $job.activeDeadlineSeconds }}
+{{ end }}
```
Comment (aslafy-z) [inline application/templates/cronjob.yaml:36]:
```suggestion
{{- end }}
```

File: application/templates/cronjob.yaml [hunk 3]
```diff
@@ -31,10 +31,17 @@ spec:
 {{ end }}
   jobTemplate:
     spec:
+{{- if $job.activeDeadlineSeconds }}
```
Comment (aslafy-z) [inline application/templates/cronjob.yaml:34]: You can use with in here

File: application/templates/cronjob.yaml [hunk 4]
```diff
@@ -31,10 +31,17 @@ spec:
...
+      activeDeadlineSeconds: {{ $job.activeDeadlineSeconds }}
+{{ end }}
       template:
         metadata:
           labels:
           {{- include "application.labels" $ | nindent 12 }}
+          {{- with $job.additionalPodAnnotations }}
```
Comment (aslafy-z) [inline application/templates/cronjob.yaml:41]:
```suggestion
          {{- with $job.podAnnotations }}
```
is enough

File: application/templates/cronjob.yaml (hunk 3 above)
Comment (AntoineFr) [reply application/templates/cronjob.yaml]: Changed in [49d195c](https://github.com/stakater/application/pull/265/commits/49d195c7bc40c21c578d5fc03c9445fa6a517d80)

File: application/templates/cronjob.yaml (hunk 4 above)
Comment (AntoineFr) [reply application/templates/cronjob.yaml:44]: I also added a property `additionalPodLabels`, now isn't it better to keep both with the same name format ?

File: application/templates/cronjob.yaml [hunk 5]
```diff
@@ -31,10 +31,19 @@ spec:
...
     spec:
+{{- with $job.activeDeadlineSeconds }}
+      activeDeadlineSeconds: {{ . }}
+{{- end }}
       template:
         metadata:
           labels:
           {{- include "application.labels" $ | nindent 12 }}
+          {{- with $job.additionalPodLabels }}
+          {{ toYaml . | nindent 12 }}
```
Comment (aslafy-z) [inline application/templates/cronjob.yaml:42]:
```suggestion
          {{- toYaml . | indent 12 }}
```
I think `{{- include "application.labels" $ | nindent 12 }}` also has an unnecessary N in Nindent.

File: application/templates/cronjob.yaml [hunk 6]
```diff
@@ -88,6 +97,14 @@ spec:
           {{- end }}
           {{- with $job.tolerations }}
           tolerations:
+{{ toYaml . | indent 12 }}
```
Comment (aslafy-z) [inline application/templates/cronjob.yaml:100]:
```suggestion
          tolerations: {{ toYaml . | nindent 12 }}
```

File: application/templates/cronjob.yaml [hunk 7]
```diff
@@ -88,6 +97,14 @@ spec:
...
+          {{- end }}
+          {{- with $job.topologySpreadConstraints }}
+          topologySpreadConstraints:
+{{ toYaml . | indent 12 }}
```
Comment (aslafy-z) [inline application/templates/cronjob.yaml:104]:
```suggestion
          topologySpreadConstraints: {{ toYaml . | nindent 12 }}
```

File: application/templates/cronjob.yaml [hunk 8]
```diff
@@ -88,6 +97,14 @@ spec:
...
+          {{- end }}
+          {{- with $job.securityContext }}
+          securityContext:
 {{ toYaml . | indent 12 }}
```
Comment (aslafy-z) [inline application/templates/cronjob.yaml:108]:
```suggestion
          securityContext: {{ toYaml . | nindent 12 }}
```

File: application/templates/cronjob.yaml (hunk 4 above)
Comment (aslafy-z) [reply application/templates/cronjob.yaml:44]: I don't really care, as you'd like

Comment (AntoineFr) [reply application/templates/cronjob.yaml:44]: I prefer to have them with the same format :)

File: application/templates/cronjob.yaml [hunk 9]
```diff
@@ -31,10 +31,19 @@ spec:
 {{ end }}
   jobTemplate:
     spec:
+{{- with $job.activeDeadlineSeconds }}
+      activeDeadlineSeconds: {{ . }}
+{{- end }}
```
Comment (aslafy-z) [inline application/templates/cronjob.yaml:36]:
```suggestion
      {{- with $job.activeDeadlineSeconds }}
      activeDeadlineSeconds: {{ . }}
      {{- end }}
```

**Review by aslafy-z — DISMISSED** [review]

**Comment by rasheedamir** [comment]
@aslafy-z should we get some tests added as well? And how about changelog file that should be required as well

**Comment by AntoineFr** [comment]
Hello @rasheedamir , I updated the changelog in 3332fa4

For the tests, this PR only adds some simple mappings so I don't know if they would add a lot of value

**Comment by rasheedamir** [comment]
@AntoineFr plz resolve conflicts! and then lets merge @aslafy-z

**Comment by AntoineFr** [comment]
@rasheedamir the conflicts are gone 🎉

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,9 @@
 
 All notable changes to this project will be documented here.
 
+### v2.2.2
```
Comment (aslafy-z) [inline CHANGELOG.md:6]: Next release will be 2.2.3. 

```suggestion
### v2.2.3
```

**Review by aslafy-z — APPROVED** [review]

**Review by aslafy-z — APPROVED** [review]

**Pull request merged**